### PR TITLE
Add offset option to cat command

### PR DESCRIPTION
**Pull request by frrist**

### CODE REVIEW

**Comment by frrist** [comment]
Discussed in issue #4085 & #3578

File: core/commands/cat.go [hunk 1]
```diff
@@ -103,16 +116,26 @@ var CatCmd = &cmds.Command{
...
 		}
+		if offset > int64(read.Size()) {
+			offset = offset - int64(read.Size())
+			continue
+		}
+		count, err := read.Seek(offset, io.SeekStart)
+		if err != nil {
+			return nil, 0, err
+		}
+		offset = offset - count
```
Comment (whyrusleeping) [inline core/commands/cat.go]: shouldnt this always result in offset being 0?

File: test/sharness/t0040-add-and-cat.sh [hunk 2]
```diff
@@ -46,11 +46,44 @@ test_add_cat_file() {
...
+  test_expect_success "ipfs cat with offset succeeds" '
+    ipfs cat --offset 10 "$HASH" >actual
+  '
+
+  test_expect_success "ipfs cat from offset output looks good" '
+    echo "ds!" >expected &&
+    test_cmp expected actual
+  '
+
+  test_expect_success "ipfs cat multipule hashes with offset succeeds" '
```
Comment (whyrusleeping) [inline test/sharness/t0040-add-and-cat.sh]: small typo

**Comment by whyrusleeping** [comment]
a couple things, then LGTM.

**Comment by whyrusleeping** [comment]
right, this closes #3578, but i think for #4085 we need a `--length` flag as well. I'm fine having that be a separate PR

**Pull request merged**